drivers/usb/udc/udc_dwc2.c

```diff
@@ -853,8 +887,15 @@ static inline void dwc2_handle_rxflvl(const struct device *dev)
 			break;
 		}
 
-		if (net_buf_tailroom(buf) && evt.bcnt == ep_cfg->mps) {
-			dwc2_prep_rx(dev, buf, ep_cfg, 0);
+		if (((evt.bcnt % USB_MPS_EP_SIZE(ep_cfg->mps)) == 0) &&
```

Comment:
This PR does not change the use of the MPS value in the other drivers. We could consider ignoring it for the full speed controllers, but there are few drivers for full and high speed controllers, I think it is better to unify it and not use mps value directly. I also want to introduce udc_mps_ep_size() to get the size field from the endpoint configuration structure, see https://github.com/jfischer-no/zephyr/commit/1d84372d5d95a0b6864dc266afe84b5cb5be4c53#diff-e35b5db282dc4b82092896f84fea69d5be994a93c813eb07cad87f55b33c2523, [patch](https://github.com/jfischer-no/zephyr/commit/1d84372d5d95a0b6864dc266afe84b5cb5be4c53.patch)

Reply:
Other drivers do not set `caps.high_bandwidth` and therefore won't ever get mps that is greater than 1024. Only when the drivers do set `high_bandwidth` capability it is necessary to differentiate between wMaxPacketSize value and actual endpoint size.

I think the unification can come later.

Reply:
I do not see why it should be later, you just have to cherrypick. This is the right time to unify it and also set a good example for new drivers.

Reply:
Cherry picked your commit and pushed. After the push, GitHub reported conflict in stm32 driver and for that reason I did rebase on top of main. Please review.